db/generateSeedScript.js

```diff
@@ -0,0 +1,111 @@
+
+// style adjustment pricing 
+var priceAdjust = [-20, 20, -10, -10, -10, -10, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 10, 10, 10, 10, 10, 20, 20]
+
+var sizes = {};
+sizes["Men's Shoe"] = ['6','6.5','7','7.5','8','8.5','9','9.5','10','10.5','11','11.5','12','12.5','13','14','15'];
+sizes["Women's Shoe"]  = ['5','5.5','6','6.5','7','7.5','8','8.5','9','9.5','10','10.5','11','11.5','12'];
+
+var chars = ['A','B','C','D','E','F','G','H','I','J','K','L','M','N','O','P','Q','R','S','T','U','V','W','X','Y','Z']
+
+var inventory = [0,0,0,0,0,0,0,0,0,0,1,1,2,2,3,3,4,4,5,5,6,6,7,7,8,8,9,9,10,10]
+
+// a more flexible randomizer that takes arrays and numbers
+var rnd = n => Array.isArray(n) ? n[ rnd(n.length) - 1] : Math.floor( Math.random() * n ) + 1
+
+// generate a random style code ( "AB-1234-styleId" )
+var getStyleCode = id => rnd(chars) + rnd(chars) + (rnd(9000) + 1000) + "-" + id;
+
+// from the base price of a product, returns an array containing plausible base & sale prices for a style 
+var getStylePrice = base => {
+  var adjust = base + rnd( priceAdjust ); // set the adjusted price for this style
+  return rnd(2) == 2 
+          ? [adjust, adjust] // 50% chance of a sale
+          : [adjust, adjust - rnd( adjust / 4 ) - .03] // up to 25% off; sale prices end in .97
+}
+
+// for each of the 100 products, give it a range of style options
+var generateSeedScript = (min, max) => {
+  var styleId = 0, productList = '\n-- products\n', styleList = '\n-- styles\n', inventoryList = '\n-- inventory\n';
+
+  products.forEach( (product, id) => {
+    var productURI = encodeURIComponent( product.toLowerCase().split( ' ' ).join( '-' ) );
```

Comment:
I've never used encodeURIComponent. Just learned about it just now as a result so that's awesome.

I'm curious, which characters in the array of products is it meant to encode. It seemed like it's composed of alphabetical characters and dashes which don't get encoded.

Reply:
It was mostly a safety measure - the first item had an apostrophe in it, so I just wanted to make sure everything was a safe character to go into a URL.

Reply:
okay awesome